__tests__/Unit/hooks/tasksApi.test.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+    failedAddNewTaskHandler,
+    failedGetTasks,
+    failedUpdateTaskHandler,
+} from '../../../__mocks__/handlers/tasks.handler';
+
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+    server.listen();
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+function Wrapper({
+    children,
+}: PropsWithChildren<Record<string, never>>): JSX.Element {
+    return <Provider store={store()}>{children}</Provider>;
+}
+
+describe('useGetAllTasksQuery()', () => {
+    test('returns all tasks', async () => {
+        const { result, waitForNextUpdate } = renderHook(
+            () => useGetAllTasksQuery(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const initialResponse = result.current;
+        expect(initialResponse.data).toBeUndefined();
+        expect(initialResponse.isLoading).toBe(true);
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current;
+        const tasksData = nextResponse.data;
+        expect(tasksData).not.toBeUndefined();
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.isSuccess).toBe(true);
+    });
+
+    test('should fail to return all tasks with error', async () => {
+        server.use(failedGetTasks);
+        const { result, waitForNextUpdate } = renderHook(
+            () => useGetAllTasksQuery(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const initialResponse = result.current;
+        expect(initialResponse.isLoading).toBe(true);
+        expect(initialResponse.data).toBeUndefined();
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current;
+        expect(nextResponse.isError).toBe(true);
+        expect(nextResponse.error).toHaveProperty('status', 500);
+    });
+});
+
+describe('useAddTaskMutation()', () => {
+    test('creates a new task', async () => {
+        const newTask = {
+            title: 'SSG and SSR config to be setup ',
+            type: 'feature',
+            status: 'AVAILABLE',
+            percentCompleted: 0,
+            priority: 'TBD',
+            github: {
+                issue: {
+                    status: 'open',
+                    id: 697061944,
+                },
+            },
+        };
+        const { result, waitForNextUpdate } = renderHook(
+            () => useAddTaskMutation(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const [addTask, initialResponse] = result.current;
+        expect(initialResponse.isLoading).toBe(false);
+        expect(initialResponse.data).toBeUndefined();
+
+        act(() => {
+            addTask(newTask);
+        });
+
+        const loadingResponse = result.current[1];
+        expect(loadingResponse.isLoading).toBe(true);
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current[1];
```

Comment:
do we have an array as a result of task creation? why are we specifically selecting the 2nd value from the response

Reply:
Yes, `useMutation` returns an array
Kindly read https://redux-toolkit.js.org/rtk-query/usage/mutations#frequently-used-mutation-hook-return-values